File: Selenium/Activities/Activity8.py
# Input Events #1
# Using Selenium:
# Open a new browser to https://training-support.net/webelements/mouse-events
# Get the title of the page and print it to the console.
# On the page, perform:
# Left click on the Cargo.lock button, move the cursor to the Cargo.toml button and then click it. Print the confirmation text at the end of the sequence.
# Double click on the src button. Then right click on the target button and select open. Print the confirmation text at the end of the sequence.
# Close the browser.
from tokenize import String
from selenium import webdriver
from selenium.webdriver.common.by import By
driver = webdriver.Firefox()
driver.get("https://training-support.net/webelements/mouse-events")
print("The title of the page is: ", driver.title)
cargolock = driver.find_element(By.XPATH, "//h1[contains(text(), 'Cargo.lock')]")
cargotoml = driver.find_element(By.XPATH, "//h1[contains(text(), 'Cargo.toml')]")
src = driver.find_element(By.XPATH, "//h1[contains(text(), 'src')]")
target = driver.find_element(By.XPATH, "//h1[contains(text(), 'target')]")
actions = webdriver.ActionChains(driver)
actions.click(cargolock).pause(1).move_to_element(cargotoml).click(cargotoml).pause(1).perform()
confirmmsg = driver.find_element(By.XPATH, "//p[@id=\"result\"]").text
print("The confirmation message is: ", confirmmsg)
actions.move_to_element(src).pause(1).double_click(src).pause(1).move_to_element(target).context_click(target).perform()
openoption = driver.find_element(By.XPATH, "//span[contains(text(), 'Open')]")
actions.click(openoption).pause(1).perform()
finalmsg = driver.find_element(By.XPATH, "//p[@id=\"result\"]").text
print("The final message is: ", finalmsg)
driver.quit()

File: Selenium/Activities/Activity8.java
//Input Events #1
//Using Selenium:
//Open a new browser to https://training-support.net/webelements/mouse-events
//Get the title of the page and print it to the console.
//On the page, perform:
//Left click on the Cargo.lock button, move the cursor to the Cargo.toml button and then click it. Print the confirmation text at the end of the sequence.
//Double click on the src button. Then right click on the target button and select open. Print the confirmation text at the end of the sequence.
//Close the browser.

package activities;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity8 {
	public static void main(String[] args) {
		WebDriver driver = new FirefoxDriver();
		driver.get("https://training-support.net/webelements/mouse-events");
		System.out.println("The title of the page is: " + driver.getTitle());
		WebElement cargolock = driver.findElement(By.xpath("//h1[contains(text(), 'Cargo.lock')]"));
		WebElement cargoToml = driver.findElement(By.xpath("//h1[contains(text(), 'Cargo.toml')]"));
		WebElement src = driver.findElement(By.xpath("//h1[contains(text(), 'src')]"));
		WebElement target = driver.findElement(By.xpath("//h1[contains(text(), 'target')]"));
		Actions builder = new Actions(driver);
		builder.click(cargolock).pause(1000).moveToElement(cargoToml).pause(1000).click(cargoToml).build().perform();
		String confirmmessage = driver.findElement(By.xpath("//p[@id=\"result\"]")).getText();
		System.out.println("The confirmation message is: " + confirmmessage);
		builder.moveToElement(src).pause(1000).doubleClick(src).pause(1000).moveToElement(target).contextClick(target).build().perform();
		WebElement openoption = driver.findElement(By.xpath("//span[contains(text(), 'Open')]"));
		builder.click(openoption).pause(1000).build().perform();
		String actionmessage = driver.findElement(By.xpath("//p[@id=\"result\"]")).getText();
		System.out.println("The final message is: " + actionmessage);
		driver.quit();
	}

}
